deepmd/pd/train/training.py

```
@@ -0,0 +1,1249 @@
+
+        # Tensorboard
+        self.enable_tensorboard = training_params.get("tensorboard", False)
+        self.tensorboard_log_dir = training_params.get("tensorboard_log_dir", "log")
+        self.tensorboard_freq = training_params.get("tensorboard_freq", 1)
+        self.enable_profiler = training_params.get("enable_profiler", False)
+        self.profiling = training_params.get("profiling", False)
+        self.profiling_file = training_params.get("profiling_file", "timeline.json")
+
+    def run(self):
+        fout = (
+            open(
```

Ruff (0.8.0)

Use a context manager for opening files

(SIM115)